src/test/Components/CreateImageWizardV2/steps/FirstBoot/Firstboot.test.tsx

```diff
@@ -0,0 +1,116 @@
+import { screen } from '@testing-library/react';
+import { userEvent } from '@testing-library/user-event';
+
+import {
+  CREATE_BLUEPRINT,
+  FIRST_BOOT_SERVICE_DATA,
+} from '../../../../../constants';
+import { File as ImageBuilderFile } from '../../../../../store/imageBuilderApi';
+import { clickNext } from '../../../../testUtils';
+import {
+  blueprintRequest,
+  clickRegisterLater,
+  enterBlueprintName,
+  interceptBlueprintRequest,
+  render,
+} from '../../wizardTestUtils';
+
+import '@testing-library/jest-dom';
+
+jest.mock('@redhat-cloud-services/frontend-components/useChrome', () => ({
+  useChrome: () => ({
+    auth: {
+      getUser: () => {
+        return {
+          identity: {
+            internal: {
+              org_id: 5,
+            },
+          },
+        };
+      },
+    },
+    isBeta: () => true,
+    isProd: () => true,
+    getEnvironment: () => 'prod',
+  }),
+}));
+
+const goToFirstBootStep = async () => {
+  const bareMetalCheckBox = await screen.findByRole('checkbox', {
+    name: /bare metal installer checkbox/i,
+  });
+  await userEvent.click(bareMetalCheckBox);
+  await clickNext(); // Registration
+  await clickRegisterLater();
+  await clickNext(); // OpenSCAP
+  await clickNext(); // File System
+  await clickNext(); // Custom repositories
+  await clickNext(); // Additional packages
+  await clickNext(); // First Boot
+};
+
+const goToReviewStep = async () => {
+  await clickNext(); // Details
+  await enterBlueprintName();
+  await clickNext(); // Review
+};
+
+const SCRIPT = `#!/bin/bash
+systemctl enable cockpit.socket`;
+
+const getFirstbootData = (): ImageBuilderFile[] => [
+  {
+    path: '/etc/systemd/system/custom-first-boot.service',
+    data: FIRST_BOOT_SERVICE_DATA,
+    data_encoding: 'base64',
+    ensure_parents: true,
+  },
+  {
+    path: '/usr/local/sbin/custom-first-boot',
+    data: btoa(SCRIPT),
+    data_encoding: 'base64',
+    mode: '0774',
+    ensure_parents: true,
+  },
+];
+
+describe('First Boot step', () => {
+  test('should render First Boot step', async () => {
+    await render();
+    await goToFirstBootStep();
+    expect(screen.getByText('First boot configuration')).toBeInTheDocument();
+  });
+  describe('validate first boot request ', () => {
+    test('should validate first boot request', async () => {
```

Comment:
@amirfefer would you mind refactoring this a bit by extracting the steps into small functions with clear names?

For instance this might be something like:
```typescript
await render();
await goToFirstBootStep();
await addShellScript();
await goToReviewStep();
...
```